### PR TITLE
[IMP] Timesheets: Back2basics

### PR DESCRIPTION
**Pull request by jbm-odoo**
Add internal project to ease the onboarding
Move timer to enterprise

taskID: 2217930

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_timesheet/models/hr_timesheet.py [hunk 1]
```diff
@@ -234,4 +234,12 @@ def _add_timesheet_time(self, minutes_spent):
...
     def _action_interrupt_user_timers(self):
-        self.action_timer_stop()
\ No newline at end of file
+        self.action_timer_stop()
+
+    def unlink(self):
+        res = super(AccountAnalyticLine, self).unlink()
+        self.env['timer.timer'].search([
+            ('res_model', '=', self._name),
+            ('res_id', 'in', self.ids)
```
[the remark this answers is not in the document]
Comment (jbm-odoo) [reply addons/hr_timesheet/models/hr_timesheet.py]: Yes, I made some tests, and it's working :)

File: addons/timer/static/src/js/timer_mixin.js [hunk 2]
```diff
@@ -39,9 +39,9 @@ odoo.define('timer.Timer', function (require) {
 
             timer.addTime(
                 moment.utc(
-                    moment.utc(serverTime)
+                    Math.abs(moment.utc(serverTime)
```
[the remark this answers is not in the document]
Comment (jbm-odoo) [reply addons/timer/static/src/js/timer_mixin.js]: It's not really a bug. After my commit, we can add negative time to timer and this use case wasn't possible before this commit.

File: addons/hr_timesheet/__init__.py [hunk 3]
```diff
@@ -5,3 +5,43 @@
...
+        if not admin.employee_ids.filtered(lambda e: e.company_id == company):
+            continue
+        company.env['account.analytic.line'].create([{
+            'name': _("Analysis"),
+            'user_id': admin.id,
+            'date': fields.datetime.today(),
+            'unit_amount': 0,
+            'project_id': internal_project.id,
+            'task_id': task.id,
+        } for task in tasks])
```
Comment (tivisse) [inline addons/hr_timesheet/__init__.py:47]: Being peaky over here. Would be better to reduce the number of create. 
```suggestion
    tasks_vals = []
    for company in env['res.company'].search([('id', 'not in', company_ids_with_internal)]):
        company = company.with_company(company)
        internal_project = company.env['project.project'].create({
            'name': _('Internal'),
            'allow_timesheets': True,
            'company_id': company.id,
        })
        admin = env.ref('base.user_admin', raise_if_not_found=False)
        if not admin:
            continue
        tasks_vals.append[{
            'name': _('Training'),
            'project_id': internal_project.id,
            'company_id': company.id,
        }, {
            'name': _('Meeting'),
            'project_id': internal_project.id,
            'company_id': company.id,
        }]

    tasks = self.env['project.task'].create(tasks_vals)

    if not admin.employee_ids.filtered(lambda e: e.company_id == company):
        continue
    company.env['account.analytic.line'].create([{
        'name': _("Analysis"),
        'user_id': admin.id,
        'date': fields.datetime.today(),
        'unit_amount': 0,
        'project_id': internal_project.id,
        'task_id': task.id,
    } for task in tasks])
```

Comment (tivisse) [reply addons/hr_timesheet/__init__.py:47]: And we can go further by doing the same with projects, I let you handle this ;)

**Comment by tivisse** [comment]
@robodoo r+ merge

**Pull request merged**